simln-lib/src/sim_node.rs

```diff
@@ -848,6 +848,9 @@ pub struct SimGraph {
     /// Optional writer to flush htlc forwards to disk.
     writer: Option<Arc<Mutex<BatchedWriter>>>,
 
+    /// Custom records to add to htlcs.
+    custom_records: CustomRecords,
```

Comment:
nit: `default_custom_records` and note that these will be added to the first outgoing HTLC in a payment.